### PR TITLE
CDAP-17601 rename the display name of staging bucket locatin to dataset location

### PR DESCRIPTION
**Pull request by seanzhougoogle**

### CODE REVIEW

File: docs/bigquery-cdcTarget.md [hunk 1]
```diff
@@ -40,6 +40,11 @@ Properties
 **Project ID**: Project of the BigQuery dataset. When running on a Dataproc cluster, this can be left blank,
 which will use the project of the cluster.
 
+**Location**: The location where the BigQuery dataset and GCS staging gcs bucket will get created. For example, 'us-east1'
```
Comment (avirmani) [inline docs/bigquery-cdcTarget.md]: Can omit 'gcs'. It can just be 'GCS staging bucket'

File: docs/bigquery-cdcTarget.md [hunk 2]
```diff
@@ -40,6 +40,11 @@ Properties
 **Project ID**: Project of the BigQuery dataset. When running on a Dataproc cluster, this can be left blank,
 which will use the project of the cluster.
 
+**Location**: The location where the BigQuery dataset and GCS staging gcs bucket will get created. For example, 'us-east1' 
+for regional bucket, 'us' for multi-regional bucket. Complete list of available GCS locations can be found here 
+https://cloud.google.com/bigquery/docs/locations. This value is ignored if an existing GCS bucket is specified as staging
```
Comment (albertshau) [inline docs/bigquery-cdcTarget.md]: specified, as the staging bucket ...

File: docs/bigquery-cdcTarget.md [hunk 3]
```diff
@@ -40,6 +40,11 @@ Properties
 **Project ID**: Project of the BigQuery dataset. When running on a Dataproc cluster, this can be left blank,
 which will use the project of the cluster.
 
+**Location**: The location where the BigQuery dataset and GCS staging gcs bucket will get created. For example, 'us-east1' 
+for regional bucket, 'us' for multi-regional bucket. Complete list of available GCS locations can be found here
```
Comment (albertshau) [inline docs/bigquery-cdcTarget.md]: A complete list of available locations can be found at https://...

File: src/main/java/io/cdap/delta/bigquery/BigQueryTarget.java [hunk 4]
```diff
@@ -200,9 +200,10 @@ private static String stringifyPipelineId(DeltaPipelineId pipelineId) {
     private String stagingBucket;
 
     @Nullable
-    @Description("The location where the staging gcs bucket will get created. "
-      + "This value is ignored if the bucket already exists.")
-    protected String stagingBucketLocation;
+    @Description("The location where the Bigquery dataset and GCS staging bucket will be created. Bigquery dataset " +
```
Comment (avirmani) [inline src/main/java/io/cdap/delta/bigquery/BigQueryTarget.java]: BigQuery

**Review by avirmani — COMMENTED** [review]
LGTM - some minor doc comments

File: src/main/java/io/cdap/delta/bigquery/BigQueryTarget.java [hunk 5]
```diff
@@ -200,9 +200,10 @@ private static String stringifyPipelineId(DeltaPipelineId pipelineId) {
...
+      "and GCS Staging Bucket must be in the same location. This value is ignored if an existing GCS bucket is " +
+      "specified as staging bucket and the Bigquery dataset will be created in the same location as that bucket")
+    protected String location;
```
Comment (albertshau) [inline src/main/java/io/cdap/delta/bigquery/BigQueryTarget.java]: renaming this is backwards incompatible (for ex: upgrading a replicator will cause the old value to be ignored), is that ok?

If not, there needs to be a little more complicated logic where this new one is introduced, stagingBucketLocation uses 
 the 'hidden' widget-type, and both are checked in the backend.

Comment (seanzhougoogle) [reply src/main/java/io/cdap/delta/bigquery/BigQueryTarget.java]: good point, then I think we don't need to bother renaming the the variable name.

File: docs/bigquery-cdcTarget.md (hunk 1 above)
Comment (seanzhougoogle) [reply docs/bigquery-cdcTarget.md]: done :)

File: docs/bigquery-cdcTarget.md (hunk 2 above)
Comment (seanzhougoogle) [reply docs/bigquery-cdcTarget.md]: done :)

File: src/main/java/io/cdap/delta/bigquery/BigQueryTarget.java (hunk 4 above)
Comment (seanzhougoogle) [reply src/main/java/io/cdap/delta/bigquery/BigQueryTarget.java]: done :)

**Comment by avirmani** [comment]
LGTM

File: docs/bigquery-cdcTarget.md [hunk 6]
```diff
@@ -40,6 +40,11 @@ Properties
 **Project ID**: Project of the BigQuery dataset. When running on a Dataproc cluster, this can be left blank,
 which will use the project of the cluster.
 
+**Location**: The location where the BigQuery dataset and GCS staging bucket will get created. For example, 'us-east1' 
+for regional bucket, 'us' for multi-regional bucket. A complete list of available locations can be found at 
+https://cloud.google.com/bigquery/docs/locations. This value is ignored if an existing GCS bucket is specified, as staging
```
Comment (albertshau) [inline docs/bigquery-cdcTarget.md]: as staging -> as the staging

**Review by albertshau — APPROVED** [review]

**Pull request merged**